feat: implement like in compare util

GursheenK opened the pull request:
**Changes**
- Added functions for evaluating **like** and **not like** operators in the compare util.
- Added a simple test for checking if `evaluate_filters` works for filters including these operators.

Closes #22007

`no-docs`

GursheenK committed:
feat: implement like & not like in compare util

GursheenK committed:
refactor: remove operator check for like in recorder

ankush commented on frappe/utils/data.py:
```diff
@@ -1707,6 +1707,8 @@ def get_url_to_report_with_filters(name, filters, report_type=None, doctype=None
 	"<=": operator.le,
 	"not None": lambda a, b: a is not None,
 	"None": lambda a, b: a is None,
+	"like": lambda a, b: operator.contains(a.strip("%"), b.strip("%")),
+	"not like": lambda a, b: not operator.contains(a.strip("%"), b.strip("%")),
```
@surajshetty3416 just a question on design

Should we support SQL like `like` filtering where prefix/suffix match is supported?

ankush committed:
fix: Support SQL like `LIKE` filter

Other changes:
- Ignore empty doctype in filter creator
- Simplified recorder filter evals
